torch/distributed/launch.py

```diff
@@ -254,7 +254,7 @@ def main():
         process.wait()
         if process.returncode != 0:
             raise subprocess.CalledProcessError(returncode=process.returncode,
-                                                cmd=cmd)
+                                                cmd=process.args)
```

Comment:
hmm, is this correct? It looks like the original code relies on the for loop above to modify cmd, and then use the modified `cmd` here. E.g., it is expected to provide a `--local_rank` to here, which is only added in the for loop above.

Reply:
The modified cmd (including local_rank) is then passed to the process, so it is part of `process.args`.

The problem with the current code is that on error in process with `local_rank=0` the exception message shows `local_rank=1` (assuming two GPUs).

Sorry for jumping in btw, I was just about to open the same bug (after spending more than a day trying to debug the wrong process), when I noticed this MR.